packages/edit-site/src/utils/use-actual-current-theme.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { useState, useEffect } from '@wordpress/element';
+import { addQueryArgs } from '@wordpress/url';
+
+const ACTIVE_THEMES_URL = '/wp/v2/themes?status=active';
+
+export function useActualCurrentTheme() {
+	const [ currentTheme, setCurrentTheme ] = useState();
+
+	useEffect( () => {
+		// Set the `wp_theme_preview` to empty string to bypass the createThemePreviewMiddleware.
+		const path = addQueryArgs( ACTIVE_THEMES_URL, {
+			context: 'edit',
+			wp_theme_preview: '',
+		} );
+
+		apiFetch( { path } ).then( ( activeThemes ) =>
+			setCurrentTheme( activeThemes[ 0 ] )
```

Comment:
Does it make sense to handle a request error here? Wondering if it's safe to access `activeThemes[ 0 ]` here.

Reply:
Handled the request error by https://github.com/WordPress/gutenberg/pull/57588/commits/2f87b343b52c3488b46922334ced1268841d5c3c.

> Wondering if it's safe to access activeThemes[ 0 ] here.

It should be safe as we also access the value [here](https://github.com/WordPress/gutenberg/blob/9d632c527a4329ed87379c47b89c1ce3feda971c/packages/core-data/src/resolvers.js#L316).